ql/cashflows/inflationcoupon.cpp

```diff
@@ -70,13 +70,21 @@ namespace QuantLib {
     Real InflationCoupon::accruedAmount(const Date& d) const {
         if (d <= accrualStartDate_ || d > paymentDate_) {
             return 0.0;
-        } else {
-            return nominal() * rate() *
+        }else if (tradingExCoupon(d)) {
+            return -1*nominal() * rate() *
             dayCounter().yearFraction(accrualStartDate_,
                                       std::min(d, accrualEndDate_),
```

Comment:
Thanks!  But I think the dates are incorrect here?  This should be
```suggestion
            return -nominal() * rate() *
            dayCounter().yearFraction(d,
                                      std::max(d, accrualEndDate_),
```
as in the fixed and floating coupons, otherwise it's going to return a negative full coupon.
It would be nice to have a test case, if you have the time to write it...

Reply:
you're absolutely right, good catch my bad.
this is the one on the fixed rate coupon
```
   Real FixedRateCoupon::accruedAmount(const Date& d) const {
        if (d <= accrualStartDate_ || d > paymentDate_) {
            // out of coupon range
            return 0.0;
        } else if (tradingExCoupon(d)) {
            return -nominal()*(rate_.compoundFactor(d,
                                                    std::max(d, accrualEndDate_),
                                                    refPeriodStart_,
                                                    refPeriodEnd_) - 1.0);
        } else {
            // usual case
            return nominal()*(rate_.compoundFactor(accrualStartDate_,
                                                   std::min(d,accrualEndDate_),
                                                   refPeriodStart_,
                                                   refPeriodEnd_) - 1.0);
        }
    }
```

and floater rate coupon:
```
Real FloatingRateCoupon::accruedAmount(const Date& d) const {
        if (d <= accrualStartDate_ || d > paymentDate_) {
            // out of coupon range
            return 0.0;
        } else if (tradingExCoupon(d)) {
            return -nominal() * rate() *
                   dayCounter().yearFraction(d, std::max(d, accrualEndDate_),
                                             refPeriodStart_, refPeriodEnd_);
        } else {
            // usual case
            return nominal() * rate() *
                dayCounter().yearFraction(accrualStartDate_,
                                          std::min(d, accrualEndDate_),
                                          refPeriodStart_,
                                          refPeriodEnd_);
        }
    }
```